crypto/src/main/java/org/hyperledger/besu/crypto/PublicKey.java

```diff
@@ -0,0 +1,121 @@
+package org.hyperledger.besu.crypto;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.MutableBytes;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.math.ec.ECPoint;
+import org.bouncycastle.math.ec.FixedPointCombMultiplier;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class PublicKey implements java.security.PublicKey {
+
+    public static final int BYTE_LENGTH = 64;
```

Comment:
We shouldn't presume a key length.  What if secp256r1 is un-recommened and we have to go to secp384r1, or secp521r1?  or any of the sect series?

Reply:
I would suggest to do that in a seperate PR. I've looked a bit into it and there are some calculation, e.g. to recover the public key, which are affected by that too. I think a seperate (and much smaller) PR is much easier to review such a bit more difficult change.
I will do that after this one is merged.